diff --git a/Rakefile b/Rakefile
index 223186a..fe8fe3f 100644
--- a/Rakefile
+++ b/Rakefile
@@ -6,7 +6,7 @@ USERNAME   = 'puppet'
 PASSWORD   = 'figaro'
 URLPREFIX  = 'http://localhost:9090'
 PREFS      = File.expand_path('~/.courseware.json')
-REPOSITORY = 'git@github.com:puppet-training/courseware-fundamentals.git'
+REPOSITORY = 'puppet-training/courseware-fundamentals'
 
 ## For watermarks
 AUTHOR     = 'Puppet Labs'
@@ -80,15 +80,28 @@ end
 
 desc "Ensure that the Courseware repo is correct and up to date."
 task :courseware => ['init'] do
+
+  origin = `git config --get remote.origin.url`.chomp
+
+  unless origin == "git@github.com:#{REPOSITORY}" or
+         origin == "git@github.com:#{REPOSITORY}.git" or
+         origin == "https://github.com/#{REPOSITORY}" or
+         origin == "https://github.com/#{REPOSITORY}.git"
+    raise "Your origin does not appear to be configured correctly (got #{origin})."
+  end
+
+  courseware = "git@github.com:#{REPOSITORY}.git" if origin.match(/^git/)
+  courseware = "https://github.com/#{REPOSITORY}.git" if origin.match(/^https/)
+    
   unless system('git config --get remote.courseware.url')
     # Add the remote if it doesn't already exist
-    unless system("git remote add courseware #{REPOSITORY}")
+    unless system("git remote add courseware #{courseware}")
       raise "Could not add the 'courseware' remote."
     end
   end
 
-  unless `git config --get remote.courseware.url`.chomp == REPOSITORY
-    raise "Your repository does not appear to be configured correctly. Your 'courseware' remote should be #{REPOSITORY}"
+  unless `git config --get remote.courseware.url`.chomp == courseware
+    raise "Your repository does not appear to be configured correctly. Your 'courseware' remote should be #{courseware}"
   end
 
   unless system("git fetch courseware")
